src/interfaces.ts

```diff
@@ -204,8 +205,8 @@ export interface CompilerPlugin {
     beforeFileParse?: (source: SourceObj) => void;
     afterFileParse?: (file: BscFile) => void;
     afterFileValidate?: (file: BscFile) => void;
-    beforeFileTranspile?: (entry: TranspileObj) => void;
-    afterFileTranspile?: (entry: TranspileObj) => void;
+    beforeFileTranspile?: PluginHandler<BeforeFileTranspileEvent>;
```

Comment:
Breaking public API

Reply:
Unless I'm missing something, it should still be the same public API. 

Here's the definition of `PluginHandler`. 
```
export type PluginHandler<T> = (event: T) => void;
```

Also, the `BeforeTranspileEvent` interface includes both properties from `TranspileObj`, so that also is still API compatible since _we_ produce the event for the subscribers (i.e. users don't need to create the events). 

```typescript
export interface BeforeFileTranspileEvent {
    file: BscFile;
    outputPath: string;
    astEditor: AstEditor;
}

export interface TranspileObj {
    file: BscFile;
    outputPath: string;
}
```

Reply:
Ah ok